toolkit/popup/src/main/java/com/arcgismaps/toolkit/popup/Popup.kt

```diff
@@ -91,26 +65,63 @@ private data class PopupState(@Stable val popup: Popup)
  *
  * @since 200.5.0
  */
+@Deprecated(
+    message = "Maintained for binary compatibility. Use the overload that uses the PopupState object.",
+    level = DeprecationLevel.HIDDEN
+)
 @Composable
 public fun Popup(popup: Popup, modifier: Modifier = Modifier) {
+    val scope = rememberCoroutineScope()
     val stateData = remember(popup) {
-        PopupState(popup)
+        PopupState(
+            popup,
+            scope,
+            // Ignore the UtilityAssociationsFormElement as it is not supported with this API
+            ignoreList = setOf(
+                UtilityAssociationsPopupElement::class.java
+            )
+        )
+    }
+
+    if (stateData.getActivePopupStateData().initialEvaluation.value) {
+        Popup(stateData, modifier, showCloseIcon = false)
     }
-    Popup(stateData, modifier)
 }
 
 /**
- * Maintain list of attachments outside of SDK
- * https://devtopia.esri.com/runtime/apollo/issues/681
+ * A composable Popup toolkit component that enables users to see Popup content in a
+ * layer that have been configured externally.
+ *
+ * Popups may be configured in the [Web Map Viewer](https://www.arcgis.com/home/webmap/viewer.html)
+ * or [Fields Maps Designer](https://www.arcgis.com/apps/fieldmaps/)).
+ *
+ * Note : Even though the [Popup] class is not stable, there exists an internal mechanism to
+ * enable smart recompositions.
+ *
+ * @param popupState The [PopupState] object that holds the state of the Popup.
+ * @param modifier The [Modifier] to be applied to layout corresponding to the content of this
+ * Popup.
+ * @param onDismiss Callback that is invoked when the user clicks the close icon in the top app bar.
+ * @param showCloseIcon Flag to indicate if the close icon should be shown in the top app bar. If true, the [onDismiss]
+ * callback will be invoked when the close icon is clicked. Default is true.
+ * @param isNavigationEnabled Indicates if the navigation is enabled for the popup when there are
+ * [UtilityAssociationsPopupElement]s present. When true, the user can navigate to associated features
+ * and back. If false, this navigation is disabled. Default is true
+ *
+ * @since 200.9.0
  */
-private val attachments: MutableList<PopupAttachment> = mutableListOf()
-
 @Composable
-private fun Popup(popupState: PopupState, modifier: Modifier = Modifier) {
+public fun Popup(
+    popupState: PopupState,
+    modifier: Modifier = Modifier,
+    onDismiss: () -> Unit = {},
+    showCloseIcon: Boolean = true,
+    isNavigationEnabled : Boolean = true
```

Comment:
```suggestion
    isNavigationEnabled : Boolean = true
```
For `FeatureForm`, this was meant as a way for a developer to disable the navigation in case there are geometry edits on the feature (since a form isn't concerned with geometry). Is there a similar need/requirement for a developer to disable navigation for popups?

Reply:
No, I checked the design, there is no requirement. I will remove support for it.